src/FFmpegWriter.cpp

```diff
@@ -992,7 +992,14 @@ AVStream *FFmpegWriter::add_audio_stream() {
 		throw InvalidCodec("A valid audio codec could not be found for this file.", path);
 
 	// Create a new audio stream
+#if (IS_FFMPEG_3_2 && (LIBAVFORMAT_VERSION_MAJOR < 58))
+_Pragma ("GCC diagnostic push")
+_Pragma ("GCC diagnostic ignored \"-Wdeprecated-declarations\"")
+#endif
 	AV_FORMAT_NEW_STREAM(oc, audio_codec, codec, st)
+#if (IS_FFMPEG_3_2 && (LIBAVFORMAT_VERSION_MAJOR < 58))
+_Pragma ("GCC diagnostic pop")
```

Comment:
Completely unrelated to the previous discussion about `AVFormat::Context`, a question: Is there any reason these couldn't just be,
```cxx
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```
and
```cxx
#pragma GCC diagnostic pop
```
instead? 

AIUI, `_Pragma()` exists to be used in things like macros, because you can't macro a `#keyword`-style preprocessor directive. But we're not in a macro here, we're not even in a header. This is right in the C++ source, where `#pragma` should be fine, and seems clearer to me.